### PR TITLE
feat: ToolstipsV8 for inventory items

### PR DESCRIPTION
**Pull request by kokekanon**
Creditos : kondra , v8team

Many people asked for it on Discord. I'm not sure if it'll be approved, but it'll be helpful for more than one person.

image

server conections:

option 1: 1.5  8.6
https://github.com/moviebr/TFS-1.5-Downgrades/commit/d978ead7220cc50f3602f68dfe0233e7c4b2ba92

option 2: 1.3 ? 10.98
https://github.com/OTCv8/forgottenserver/commit/7f5b4fbc08711124dec86b0fcd7bfd78dd1165c4

optiones 3 : 0.4
cc:  @ thalesduarte help

option 4 : canary

option 5:  1.4.2 repo otland
https://github.com/kokekanon/1.4.2-delete/commit/7764f3703ffba18c6a21a14bce08ea2ac15b4bee

-------
```cpp

void ProtocolGame::onRecvFirstMessage(NetworkMessage& msg)
.... 
	if (operatingSystem == CLIENTOS_OTCLIENT_WINDOWS) {
		isMehah = true;
	}
	
```

> [!IMPORTANT]
> Active features OTC
g_game.enableFeature(GameItemTooltipV8);

### PD: I'm not good at CPP.
 
image

Issues addressed:
image

image

image

image

### CODE REVIEW

**Comment by Levi999x** [comment]
Tested and working
I've used the TFS 1.4.2 changes for that

**Review by Levi999x — APPROVED** [review]

File: modules/game_interface/widgets/uiitem.lua
Comment (kokekanon) [inline modules/game_interface/widgets/uiitem.lua:1]: commits: "fix : what if the text is too long?"

image
to
image

**Comment by Fischturd** [comment]
Amazing work, works well with 1.4.2, thanks for assisting with the changes there!

**Review by rodrigopaixaorj — APPROVED** [review]

**Comment by luanluciano93** [comment]
Shouldn't `setTooltip` be within condition `if self:getItem() and self:getItem():getTooltip():len() > 0 then`? What happens if `tooltip` is null?

**Comment by kokekanon** [comment]
> Shouldn't `setTooltip` be within condition `if self:getItem() and self:getItem():getTooltip():len() > 0 then`? What happens if `tooltip` is null?

I think that .


image
is simply not displayed

I could change it to "" , 
but in v8 it has nil

**Comment by luanluciano93** [comment]
```lua
 if self:getItem() and self:getItem():getTooltip():len() > 0 then
            self:setTooltip(splitTextIntoLines(self:getItem():getTooltip(), 80))
 end
 ```
 wouldn't that solve it?

**Review by rodrigopaixaorj — APPROVED** [review]

**Review by luanluciano93 — APPROVED** [review]

**Pull request merged**